### DISCUSSION TITLE
canvaskit-wasm rendering backend support.

### DISCUSSION THREAD

**Discussion by StringKe** [pixijs/pixijs#7470] [RFC]
Is it possible to support https://www.npmjs.com/package/canvaskit-wasm to render content, so that the effect of support in nodejs will be very good. At the same time canvaskit-wasm is a package of skia, and skia can support svg output

**Comment by ivanpopelyshev** [comment]
Yes, I know people who used it with pixi. Whats your problem with it?

**Reply by StringKe** [reply]
Canvaskit is the wasm package of skia, and the api of svg canvas exists in skia. Can be used to render svg content. Include text converted to path.

If pixi supports canvaskit as a rendering backend it uses wasm, so it can be used in nodejs. node-canvas needs to be compiled.

**Reply by ivanpopelyshev** [reply]
Well yeah, but why do you even need pixi with canvaskit on nodejs, pixi canvas2d capabilities are not so good

**Reply by StringKe** [reply]
I can repackage canvaskit to provide it with svg canvas capabilities. If pixi supports canvaskit, then pixi can be used to output svg directly. At the same time, skia's svg canvas supports direct conversion of text to path. At the same time, the layout of text in skia is very good.

**Reply by ivanpopelyshev** [reply]
> If pixi supports canvaskit, then pixi can be used to output svg directly.

Wait what? Why do you need pixi to make an svg file and not render it?

I mean, why do you need pixi at all if your app doesnt have webgl environment?

**Reply by ivanpopelyshev** [reply]
Maybe you mean that SVG rendering should be made through canvaskit... However supporting canvaskit means we just .. copy its texture output to our texture :) every frame. Because canvaskit on its own cant share context with anything else.. wait, canvaskit will own the texture output?

**Reply by StringKe** [reply]
You can check canvaskit's typescript type definition. 

https://github.com/google/skia/blob/main/modules/canvaskit/npm_build/types/index.d.ts#L2629
https://skia.org/docs/user/modules/canvaskit/

I don't know if this is what you need

**Reply by StringKe** [reply]
> > If pixi supports canvaskit, then pixi can be used to output svg directly.
> 
> Wait what? Why do you need pixi to make an svg file and not render it?
> 
> I mean, why do you need pixi at all if your app doesnt have webgl environment?

I want to directly use pixi api to render jpg, png, svg and pdf at the same time. The parent project (skia) of canvaskit contains these functions.

**Comment by bigtimebuddy** [comment]
PixiJS does not yet support rendering with Node.js, but we are working on it and have a plan. Some users have hack this, but it's not something that works out of the box. However, it's on the roadmap for v6.

**Reply by StringKe** [reply]
I understand that I recommend the pixi.js team to consider using canvaskit-wasm so that both the browser and node can use the same rendering program.

**Reply by eXponenta** [reply]
Why this is not  real:
1. Need support 4 render backends (with providing a equals results ):  Canvas2D, Webgl/2, WebGPU (in future) and CanvasKit
2. Node can't to use hybrid render pipelines, some plugins require it. Resource providers will not working, becuase there are not Image, imageBitmap, dom, canvas ... etc - need to full refact of all comunitty libs and introduce a lot of polyfils.
3. Wasm. Canvas kit has huge WASM binary size and glue code. You can't embed runtime to package, same problem how and with basis. If you need use basis - you runtime size increase and 400kb. CanvasKit has runtime bundle **greater that 6mb.**!!
4. Huge architectural changes.
